packages/typescript/src/index.js

```diff
@@ -15,12 +15,19 @@ module.exports = {
     // (not pre-release) here: https://github.com/tc39/ecma262/releases
     ecmaVersion: 2020,
     sourceType: 'module',
+
+    // This enables support for linting rules that require type information. We
+    // assume that the project has a `tsconfig.json` file in the directory where
+    // ESLint is being run.
+    tsconfigRootDir: process.cwd(),
```

Comment:
I assume that we always run `eslint` from the root project here, but that may not always be the case, especially in monorepos. We may have to configure this on a per-project basis.

Reply:
Tested this in `snaps-monorepo`, and it does not work out of the box with this option. All monorepo packages need to have their own `.eslintrc`, and specify the `tsconfigRootDir`. Otherwise ESLint will complain that the files are not included in `tsconfig.json` (since they aren't in the root config).